eodatasets3/prepare/sentinel_L1C_prepare.py

```diff
@@ -0,0 +1,393 @@
+"""
+Prepare eo3 metadata for Sentinel-2 Level 1C data produced by Sinergise or esa.
+"""
+
+from pathlib import Path
+from xml.dom import minidom
+import json
+import os
+from os import listdir
+from os.path import isfile, join
+import uuid
+import click
+from typing import Tuple
+from eodatasets3 import DatasetAssembler
+from eodatasets3.ui import PathPath
+import zipfile
+
+HARDCODED = {
+    "file_format": "JPEG2000",
+    "instrument": "MSI",
+    "product_family": "level1",
+    "data_type": "Level-1C",
+    "product_type": "S2MSI1C",
+}
+
+
+SENTINEL_MSI_BAND_ALIASES = {
+    "01": "coastal_aerosol",
+    "02": "blue",
+    "03": "green",
+    "04": "red",
+    "05": "red_edge_1",
+    "06": "red_edge_2",
+    "07": "red_edge_3",
+    "08": "nir_1",
+    "8A": "nir_2",
+    "09": "water_vapour",
+    "10": "swir_1_cirrus",
+    "11": "swir_2",
+    "12": "swir_3",
+}
+
+
+def find_metadata_path(find_filename, dataset_dir):
+    for root, dir, fn in os.walk(dataset_dir):
+        if find_filename in fn:
+            return os.path.join(root, fn[fn.index(find_filename)])
+
+
+def process_product_info(product_path):
+    with open(product_path) as fp:
+        product = json.loads(fp.read())
+
+        synergise_product_name = product["name"]
+        synergise_product_id = product["id"]
+        if len(product["tiles"]) > 1:
+            raise NotImplementedError("No support for multi-tiled products yet")
+        timestamp = product["tiles"][0]["timestamp"]
+        utm_zone = product["tiles"][0]["utmZone"]
+        latitude_band = product["tiles"][0]["latitudeBand"]
+        grid_square = product["tiles"][0]["gridSquare"]
+        region_code = "%s%s%s" % (utm_zone, latitude_band, grid_square)
+
+        return {
+            "synergise_product_name": synergise_product_name,
+            "synergise_product_id": synergise_product_id,
+            "timestamp": timestamp,
+            "region_code": region_code,
+        }
+
+
+def process_metadata_xml(metadata_xml_path):
```

Comment:
All the type hints from the old prepare script seem to be removed -- across all the functions in this file.

Eg, the old one: https://github.com/GeoscienceAustralia/eo-datasets/blob/eodatasets3/eodatasets3/prepare/sentinel_sinergise_L1_prepare.py#L56

(they're especially useful here because this file has a mix of `str` and `Path` variables that are all called "path", and this can be confusing to people. .... ideally, [make them all Paths](https://github.com/GeoscienceAustralia/eo-datasets/pull/130#discussion_r595605063)!)